lib/plugins/aws/deploy/index.js

```diff
@@ -95,6 +95,9 @@ class AwsDeploy {
         .then(this.setBucketName)
         .then(this.uploadArtifacts),
 
+      'before:aws:deploy:deploy:updateStack': () => BbPromise.bind(this)
```

Comment:
@pmuens @hassankhan I think introducing a separate lifecycle event for the validation might make sense here. The AWS deploy plugin _defines_ events that it implements, but does not hook its own ones.

Add `validateTemplate` to the `lifecycleEvents` array just before `updateStack` and use `aws:deploy:deploy:validateTemplate` as hook here. This will allow plugins to hook before or after the validation and the event lifecycle follows a clear separation of concerns pattern.